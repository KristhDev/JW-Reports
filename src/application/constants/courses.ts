export const coursesMessages = {
    EXPORTED_SUCCESS: 'Has exportado tus cursos correctamente. El archivo se encuentra en su carpeta de descargas.',
    PERSON_NAME_REQUIRED: 'El nombre de la persona es requerido.',
    PERSON_NAME_MIN_LENGTH: 'El nombre de la persona debe tener al menos 2 caracteres.',
    PERSON_ABOUT_REQUIRED: 'La información de la persona es requerida.',
    PERSON_ABOUT_MIN_LENGTH: 'La información de la persona debe tener al menos 10 caracteres.',
    PERSON_ADDRESS_REQUIRED: 'La dirección es requerida.',
    PERSON_ADDRESS_MIN_LENGTH: 'La dirección debe tener al menos 10 caracteres.',
    PUBLICATION_REQUIRED: 'La publicación de estudio no puede estar vacía.',
    PUBLICATION_MIN_LENGTH: 'La publicación de estudio debe tener al menos 5 caracteres.',
    UNSELECTED: 'No hay un curso seleccionado.',
    UNSELECTED_FINISH_OR_START: 'No pudes terminar o comenzar de nuevo un curso suspendido.',
    UNSELECTED_DELETE: 'No hay un curso seleccionado para eliminar.',
    UNSELECTED_UPDATE: 'No hay un curso seleccionado para actualizar.',
    FINISHED: 'No puedes suspender o renovar un curso terminado.',
    SUSPENDED_SUCCESS: 'Has suspendido el curso correctamente.',
    RENEW_SUCCESS: 'Has renovado el curso correctamente.',
    DELETED_SUCCESS: 'Has eliminado el curso correctamente.',
    FINISHED_SUCCESS: 'Has terminado el curso correctamente.',
    RESTARTED_SUCCESS: 'Has comenzado de nuevo el curso correctamente.',
    ADDED_SUCCESS: 'Has agregado el curso correctamente.',
    UPDATED_SUCCESS: 'Has actualizado el curso correctamente.',
}