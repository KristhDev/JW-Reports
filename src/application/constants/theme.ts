import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];