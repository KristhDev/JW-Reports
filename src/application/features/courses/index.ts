export {
    default as coursesReducer,
    INIT_COURSE,
    COURSES_INITIAL_STATE,
    addCourse,
    addCourses,
    addLastLessonInCourse,
    clearCourses,
    removeCourse,
    removeCourses,
    replaceLastLessonInCourse,
    setCourseFilter,
    setCourses,
    setCoursesPagination,
    setCoursesScreenHistory,
    setHasMoreCourses,
    setIsCourseDeleting,
    setIsCourseLoading,
    setIsCoursesExporting,
    setIsCoursesLoading,
    setRefreshCourses,
    setSelectedCourse,
    updateCourse,
    updateLastLessonInCourse
} from './slice';

export * from './types';