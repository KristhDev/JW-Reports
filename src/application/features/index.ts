export * from './auth';
export * from './courses';
export * from './lessons';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './types';
export * from './ui';