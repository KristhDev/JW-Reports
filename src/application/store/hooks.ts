import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();