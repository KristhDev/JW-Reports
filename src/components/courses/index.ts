export { CourseCard } from './CourseCard';
export { CoursesList } from './CoursesList';