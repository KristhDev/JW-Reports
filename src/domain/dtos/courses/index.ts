export * from './active-suspend.dto';
export * from './create.dto';
export * from './finish-start.dto';
export * from './update.dto';