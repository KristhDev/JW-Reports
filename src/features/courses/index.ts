export {
    default as coursesReducer,
    INIT_COURSE,
    INIT_LESSON,
    addCourse,
    addCourses,
    addLesson,
    addLessons,
    clearCourses,
    removeCourse,
    removeCourses,
    removeLesson,
    removeLessons,
    setCourseFilter,
    setCourses,
    setCoursesPagination,
    setCoursesScreenHistory,
    setHasMoreCourses,
    setHasMoreLessons,
    setIsCourseDeleting,
    setIsCourseLoading,
    setIsCoursesLoading,
    setIsLessonDeleting,
    setIsLessonLoading,
    setIsLessonsLoading,
    setLessons,
    setLessonsPagination,
    setRefreshCourses,
    setRefreshLessons,
    setSelectedCourse,
    setSelectedLesson,
    updateCourse,
    updateLesson
} from './slice';