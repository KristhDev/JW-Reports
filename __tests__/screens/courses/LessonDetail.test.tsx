import React from 'react';
import { render, screen } from '@testing-library/react-native';
import dayjs from 'dayjs';

import { LessonDetail } from '../../../src/screens/courses';

import { lessonSelectedState } from '../../features/courses';

import { useCourses, useTheme } from '../../../src/hooks';

import { darkColors } from '../../../src/theme';

const setSelectedLessonMock = jest.fn();

jest.mock('../../../src/hooks/useCourses.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <LessonDetail /> screen', () => {
    (useCourses as jest.Mock).mockReturnValue({
        state: lessonSelectedState,
        finishOrStartLesson: jest.fn(),
        setSelectedLesson: setSelectedLessonMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    beforeEach(() => {
        render(<LessonDetail />);
        jest.clearAllMocks();
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render lesson data', () => {
        const selectedCourse = lessonSelectedState.selectedCourse;
        const selectedLesson = lessonSelectedState.selectedLesson;
        const nextVisit = dayjs(selectedLesson.next_lesson);
        const statusLessonText = (selectedLesson.done) ? 'Impartida' : 'Por impartir';


        const title = screen.getByTestId('title-text');
        const publication = screen.getByTestId('info-text-text');
        const statusSection = screen.getByTestId('lesson-detail-status-section');
        const descriptionSection = screen.getByTestId('lesson-detail-description-section');
        const nextVisitText = screen.getByTestId('lesson-detail-next-visit-text');
        const createDate = screen.getByTestId('lesson-detail-date-created-text');

        expect(title).toBeTruthy();
        expect(title.props.children).toBe(`CLASE DEL CURSO CON ${ selectedCourse.person_name.toUpperCase() }`);
        expect(publication).toBeTruthy();
        expect(publication.props.children).toBe(selectedCourse.publication.toUpperCase());

        expect(statusSection).toBeTruthy();
        expect(statusSection.props.children[0].props.children.join('')).toBe(`Estado de la clase: ${ statusLessonText }`);
        expect(statusSection.props.children[1].props.children.props.children).toBe((!selectedLesson.done) ? '¿Terminar clase?' : '¿Reprogramar?');

        expect(descriptionSection).toBeTruthy();
        expect(descriptionSection.props.children[0].props.children).toBe((selectedLesson.done) ? 'Se analizo:' : 'Se analizará:');
        expect(descriptionSection.props.children[1].props.children).toBe(selectedLesson.description);

        expect(nextVisitText).toBeTruthy();
        expect(nextVisitText.props.children).toBe(`${ nextVisit.format('DD') } de ${ nextVisit.format('MMMM') } del ${ nextVisit.format('YYYY') }`);

        expect(createDate).toBeTruthy();
        expect(createDate.props.children).toBe(dayjs(selectedLesson.created_at).format('DD/MM/YYYY'));
    });

    it('should not call setSelectedLesson when index of state navigation is 4', () => {
        expect(setSelectedLessonMock).not.toHaveBeenCalled();
    });
});