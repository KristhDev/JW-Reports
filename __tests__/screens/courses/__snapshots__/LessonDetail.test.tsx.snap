// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LessonDetail /> screen should to match snapshot 1`] = `
[
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "flexGrow": 1,
        "paddingBottom": 100,
      }
    }
    overScrollMode="never"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "padding": 20,
              "width": "100%",
            },
            {
              "paddingBottom": 0,
              "paddingTop": 30,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 38,
                "fontWeight": "bold",
              },
              {
                "fontSize": 24,
              },
            ]
          }
          testID="title-text"
        >
          CLASE DEL CURSO CON FRED CREMIN
        </Text>
      </View>
      <View
        style={
          [
            {
              "padding": 10,
            },
            {
              "paddingHorizontal": 20,
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#BFBFBF",
                "fontSize": 17,
                "textAlign": "center",
              },
              {
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
          testID="info-text-text"
        >
          ANIMI VOLUPTAS OMNIS
        </Text>
      </View>
      <View
        style={
          {
            "paddingHorizontal": 20,
            "paddingVertical": 16,
            "width": "100%",
          }
        }
        testID="lesson-detail-status-section"
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 0,
            }
          }
        >
          Estado de la clase: 
          Por impartir
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#A0B9E2",
                "fontSize": 19,
              }
            }
          >
            ¿Terminar clase?
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "paddingHorizontal": 20,
            "paddingVertical": 16,
            "width": "100%",
          }
        }
        testID="lesson-detail-description-section"
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Se analizará:
        </Text>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 19,
            }
          }
        >
          Nemo et et non nesciunt ipsum sed rerum vero. Nostrum quae officiis qui nihil dignissimos consectetur aperiam. Vel harum ex. Ipsum ratione necessitatibus eum sint vero enim et consequatur. Ab autem rerum numquam ut. Laboriosam nihil est exercitationem unde aut reiciendis odit.
        </Text>
      </View>
      <View
        style={
          {
            "paddingHorizontal": 20,
            "paddingVertical": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 19,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Fecha:
        </Text>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 19,
            }
          }
          testID="lesson-detail-next-visit-text"
        >
          31 de December del 2020
        </Text>
      </View>
      <Text
        style={
          {
            "bottom": 20,
            "color": "#B4B4B4",
            "fontSize": 16,
            "position": "absolute",
            "right": 20,
          }
        }
        testID="lesson-detail-date-created-text"
      >
        31/12/2020
      </Text>
    </View>
  </RCTScrollView>,
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    transparent={true}
    visible={false}
  />,
]
`;
